python/ingestion/gcs_to_bq_util.py

```diff
@@ -173,6 +173,10 @@ def load_csv_as_dataframe(gcs_bucket, filename, dtype=None, chunksize=None):
     blob.download_to_filename(local_path)
     frame = pandas.read_csv(local_path, dtype=dtype, chunksize=chunksize)
 
+    # Warning: os.remove() will remove the directory entry but will not release
+    # the file's storage until the file is no longer being used by |frame|.
+    # Double warning: This will cause an exception on Windows. See
+    # https://docs.python.org/3/library/os.html#os.remove for details.
```

Comment:
The warning about windows is just for dev environments, but the first comment flags a potential race condition/memory concern for all environments. I added this comment a while ago since the call to os.remove() has behavior that could be unexpected (I originally thought that the memory referenced by the resulting dataframe could be corrupted when the file was removed, but that is not the case in Unix). I'm not sure when this comment was removed but it seems like something that is being added back in due to a desync in my workspace. I'd like to keep it in if that's all right. 

I personally think that this is kind of a code smell and would prefer to fix it, but I think it might require a bit of a refactor.

Reply:
Oh yeah definitely fine to keep the comment, I just meant it would be nice to add something like "This shouldn't affect production environments but may be an issue for local dev on Windows"